lib/trs/fake_api_client.rb

```diff
@@ -29,112 +15,213 @@ def find_teacher(trn:, date_of_birth: "1977-02-03", national_insurance_number: n
+        completed_date: completed_date.iso8601,
+        modified_at: modified_at.utc.iso8601(3)
+      )
     end
 
-    def reset_teacher_induction(...)
+    def reset_teacher_induction(trn:, modified_at: Time.zone.now)
+      Rails.logger.info("TRSFakeAPIClient pretending to reset teacher with TRN=#{trn}'s induction")
+
+      update_induction_status(
+        trn:,
+        status: 'RequiredToComplete',
+        start_date: nil,
+        completed_date: nil,
+        modified_at: modified_at.utc.iso8601(3)
+      )
     end
 
   private
 
+    def redis_client
+      @redis_client ||= Redis.new(url: ENV.fetch('REDIS_CACHE_STORE', 'redis://localhost:6379'))
+    end
+
+    def redis_induction_key(trn)
+      "#{trn}:induction"
+    end
+
+    def update_induction_status(trn:, status:, modified_at:, start_date:, completed_date: nil)
+      payload = { 'status' => status,
+                  'startDate' => start_date,
+                  'completedDate' => completed_date,
+                  'modifiedOn' => modified_at }.transform_values { |v| (v.present?) ? v : '' }
+
+      redis_client.mapped_hmset(redis_induction_key(trn), payload)
+    end
+
+    # FIXME: Named for consistency with update_induction_status in the real client, but it's doing more
+    #        than just the status, both should probably end with _induction_details
+    def retrieve_induction_status(trn)
+      redis_client
+        .hgetall(redis_induction_key(trn))
+        .transform_values { |v| (v.present?) ? v : nil }
+    end
+
     def build_trs_teacher(trn:, date_of_birth:, national_insurance_number:)
       TRS::Teacher.new(
-        teacher_params(trn:, date_of_birth:, national_insurance_number:)
-          .merge(qts)
-          .merge(itt)
-          .merge(prohibited_from_teaching)
-          .merge(induction_status)
+        teacher_params(trn:, date_of_birth:, national_insurance_number:).tap do |tp|
+          tp.merge!(qts_data)
+          tp.merge!(itt_data)
+
+          if @is_prohibited_from_teaching
+            tp.merge!(prohibited_from_teaching_data)
+          elsif @has_alerts_but_not_prohibited
+            tp.merge!(other_alert_data)
+          end
+
+          tp.merge!(induction_data(trn))
+        end
       )
     end
 
-    def build_trs_teacher_based_on_trn_range(trn:, date_of_birth:, national_insurance_number:)
-      @induction_status = 'RequiredToComplete'
-
+    def override_data_for_special_trns(trn)
       case trn.to_i
-      when 7_000_001 then raise(TRS::Errors::QTSNotAwarded)
+      when 7_000_001 then @has_qts = false
       when 7_000_002 then raise(TRS::Errors::TeacherNotFound)
-      when 7_000_003 then raise(TRS::Errors::ProhibitedFromTeaching)
+      when 7_000_003 then @is_prohibited_from_teaching = true
       when 7_000_004 then raise(TRS::Errors::TeacherDeactivated)
+      when 7_000_005 then @has_alerts_but_not_prohibited = true
+      when 7_000_006 then nil # teacher with TRN 7000006 is seeded as an early roll out mentor
```

Comment:
These changes mean the fake API client returns data instead of raising early, so it works with both the RIAB and RECT flows.